In [1]:
import numpy as np
import pandas as pd

In [2]:
cust_rating = pd.read_csv('https://query.data.world/s/ILc-P4llUraMaYN6N6Bdw7p6kUvHnj')
cust_rating.head(10)

userID  placeID  rating  food_rating  service_rating
0  U1077   135085       2            2               2
1  U1077   135038       2            2               1
2  U1077   132825       2            2               2
3  U1077   135060       1            2               2
4  U1068   135104       1            1               2
5  U1068   132740       0            0               0
6  U1068   132663       1            1               1
7  U1068   132732       0            0               0
8  U1068   132630       1            1               1
9  U1067   132584       2            2               2

  userID  placeID  rating  food_rating  service_rating  avg_rating
0  U1077   135085       2            2               2         2.0
1  U1077   135038       2            2               1         2.0
2  U1077   132825       2            2               2         2.0
3  U1077   135060       1            2               2         2.0
4  U1068   135104       1            1               2         1.0
5  U1068   132740       0            0               0         0.0
6  U1068   132663       1            1               1         1.0
7  U1068   132732       0            0               0         0.0
8  U1068   132630       1            1               1         1.0
9  U1067   132584       2            2               2         2.0

In [3]:
cust_rating['avg_rating'] = round((cust_rating['rating'] + cust_rating['food_rating'] + cust_rating['service_rating']) / 3)
cust_rating.head(10)

userID  placeID  rating  food_rating  service_rating  avg_rating
0  U1077   135085       2            2               2         2.0
1  U1077   135038       2            2               1         2.0
2  U1077   132825       2            2               2         2.0
3  U1077   135060       1            2               2         2.0
4  U1068   135104       1            1               2         1.0
5  U1068   132740       0            0               0         0.0
6  U1068   132663       1            1               1         1.0
7  U1068   132732       0            0               0         0.0
8  U1068   132630       1            1               1         1.0
9  U1067   132584       2            2               2         2.0

In [4]:
odi_df = pd.read_csv('odi-batting.csv')
odi_df.head(10)

Country            Player  Runs   MatchDate Weekday  \
0  Afghanistan  Mohammad Shahzad   118  16-02-2010     Tue   
1  Afghanistan  Mohammad Shahzad   110  01-09-2009     Tue   
2  Afghanistan  Mohammad Shahzad   100  16-08-2010     Mon   
3  Afghanistan  Mohammad Shahzad    82  10-07-2010     Sat   
4  Afghanistan  Mohammad Shahzad    57  01-07-2010     Thu   
5  Afghanistan  Mohammad Shahzad    55  07-07-2010     Wed   
6  Afghanistan  Mohammad Shahzad    37  09-10-2010     Sat   
7  Afghanistan  Mohammad Shahzad    34  09-07-2010     Fri   
8  Afghanistan  Mohammad Shahzad    20  05-07-2010     Mon   
9  Afghanistan  Mohammad Shahzad    16  30-08-2009     Sun   

                  Ground       Versus  Balls  
0     Sharjah CA Stadium       Canada    121  
1             VRA Ground  Netherlands    111  
2  Cambusdoon New Ground     Scotland     72  
3            Hazelaarweg  Netherlands    108  
4    Sportpark Westvliet       Canada     57  
5    Sportpark Westvliet  Netherlands     63  
6   Gymkhana Club Ground        Kenya     46  
7            Hazelaarweg     Scotland     44  
8             VRA Ground        Kenya     22  
9             VRA Ground  Netherlands     28

In [5]:
odi_df['Century'] = odi_df['Runs'].apply(lambda x: 1 if x >= 100 else 0)

In [6]:
odi_df.head(10)

Country            Player  Runs   MatchDate Weekday  \
0  Afghanistan  Mohammad Shahzad   118  16-02-2010     Tue   
1  Afghanistan  Mohammad Shahzad   110  01-09-2009     Tue   
2  Afghanistan  Mohammad Shahzad   100  16-08-2010     Mon   
3  Afghanistan  Mohammad Shahzad    82  10-07-2010     Sat   
4  Afghanistan  Mohammad Shahzad    57  01-07-2010     Thu   
5  Afghanistan  Mohammad Shahzad    55  07-07-2010     Wed   
6  Afghanistan  Mohammad Shahzad    37  09-10-2010     Sat   
7  Afghanistan  Mohammad Shahzad    34  09-07-2010     Fri   
8  Afghanistan  Mohammad Shahzad    20  05-07-2010     Mon   
9  Afghanistan  Mohammad Shahzad    16  30-08-2009     Sun   

                  Ground       Versus  Balls  Century  
0     Sharjah CA Stadium       Canada    121        1  
1             VRA Ground  Netherlands    111        1  
2  Cambusdoon New Ground     Scotland     72        1  
3            Hazelaarweg  Netherlands    108        0  
4    Sportpark Westvliet       Canada     57        0  
5    Sportpark Westvliet  Netherlands     63        0  
6   Gymkhana Club Ground        Kenya     46        0  
7            Hazelaarweg     Scotland     44        0  
8             VRA Ground        Kenya     22        0  
9             VRA Ground  Netherlands     28        0

In [7]:
century_counts = odi_df.groupby('Player')['Century'].sum()
century_counts.sort_values(ascending=False, inplace=True)

In [8]:
century_counts.head()

Player
Sachin R Tendulkar     48
Ricky T Ponting        30
Sanath T Jayasuriya    28
Sourav C Ganguly       22
Herschelle H Gibbs     21
Name: Century, dtype: int64

In [9]:
odi_df['Strike Rate'] = odi_df['Runs'] / odi_df['Balls'] * 100

In [10]:
odi_df.head(10)

Country            Player  Runs   MatchDate Weekday  \
0  Afghanistan  Mohammad Shahzad   118  16-02-2010     Tue   
1  Afghanistan  Mohammad Shahzad   110  01-09-2009     Tue   
2  Afghanistan  Mohammad Shahzad   100  16-08-2010     Mon   
3  Afghanistan  Mohammad Shahzad    82  10-07-2010     Sat   
4  Afghanistan  Mohammad Shahzad    57  01-07-2010     Thu   
5  Afghanistan  Mohammad Shahzad    55  07-07-2010     Wed   
6  Afghanistan  Mohammad Shahzad    37  09-10-2010     Sat   
7  Afghanistan  Mohammad Shahzad    34  09-07-2010     Fri   
8  Afghanistan  Mohammad Shahzad    20  05-07-2010     Mon   
9  Afghanistan  Mohammad Shahzad    16  30-08-2009     Sun   

                  Ground       Versus  Balls  Century  Strike Rate  
0     Sharjah CA Stadium       Canada    121        1    97.520661  
1             VRA Ground  Netherlands    111        1    99.099099  
2  Cambusdoon New Ground     Scotland     72        1   138.888889  
3            Hazelaarweg  Netherlands    108        0    75.925926  
4    Sportpark Westvliet       Canada     57        0   100.000000  
5    Sportpark Westvliet  Netherlands     63        0    87.301587  
6   Gymkhana Club Ground        Kenya     46        0    80.434783  
7            Hazelaarweg     Scotland     44        0    77.272727  
8             VRA Ground        Kenya     22        0    90.909091  
9             VRA Ground  Netherlands     28        0    57.142857

In [11]:
pivot_df = pd.pivot_table(odi_df, index="Player", columns=['Century'], values="Strike Rate",aggfunc=np.max)

In [12]:
player_stats = pivot_df.loc[['Shahid Afridi', 'Mark V Boucher', 'Sanath T Jayasuriya','Shane R Watson']] 

player_stats

Century                       0           1
Player                                     
Shahid Afridi        400.000000  255.000000
Mark V Boucher       400.000000  216.176471
Sanath T Jayasuriya  271.428571  206.153846
Shane R Watson       300.000000  192.708333

In [13]:
indian_players = odi_df[odi_df['Country'] == "India"]
indian_players.head()

Country              Player  Runs   MatchDate Weekday  \
13601   India  Sachin R Tendulkar   200  24-02-2010     Wed   
13602   India  Sachin R Tendulkar   186  08-11-1999     Mon   
13603   India  Sachin R Tendulkar   175  05-11-2009     Thu   
13604   India  Sachin R Tendulkar   163  08-03-2009     Sun   
13605   India  Sachin R Tendulkar   152  23-02-2003     Sun   

                                   Ground        Versus  Balls  Century  \
13601          Captain Roop Singh Stadium  South Africa    147        1   
13602         Lal Bahadur Shastri Stadium   New Zealand    150        1   
13603  Rajiv Gandhi International Stadium     Australia    141        1   
13604                         AMI Stadium   New Zealand    133        1   
13605                           City Oval       Namibia    151        1   

       Strike Rate  
13601   136.054422  
13602   124.000000  
13603   124.113475  
13604   122.556391  
13605   100.662252

In [14]:
indian_players['year'] = pd.to_datetime(indian_players['MatchDate']).dt.year
indian_players.head()

C:\Users\raghu\AppData\Local\Temp\ipykernel_18792\2508509141.py:1: UserWarning: Parsing dates in DD/MM/YYYY format when dayfirst=False (the default) was specified. This may lead to inconsistently parsed dates! Specify a format to ensure consistent parsing.
  indian_players['year'] = pd.to_datetime(indian_players['MatchDate']).dt.year
C:\Users\raghu\AppData\Local\Temp\ipykernel_18792\2508509141.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  indian_players['year'] = pd.to_datetime(indian_players['MatchDate']).dt.year


Country              Player  Runs   MatchDate Weekday  \
13601   India  Sachin R Tendulkar   200  24-02-2010     Wed   
13602   India  Sachin R Tendulkar   186  08-11-1999     Mon   
13603   India  Sachin R Tendulkar   175  05-11-2009     Thu   
13604   India  Sachin R Tendulkar   163  08-03-2009     Sun   
13605   India  Sachin R Tendulkar   152  23-02-2003     Sun   

                                   Ground        Versus  Balls  Century  \
13601          Captain Roop Singh Stadium  South Africa    147        1   
13602         Lal Bahadur Shastri Stadium   New Zealand    150        1   
13603  Rajiv Gandhi International Stadium     Australia    141        1   
13604                         AMI Stadium   New Zealand    133        1   
13605                           City Oval       Namibia    151        1   

       Strike Rate  year  
13601   136.054422  2010  
13602   124.000000  1999  
13603   124.113475  2009  
13604   122.556391  2009  
13605   100.662252  2003

In [15]:
centuries_by_year = indian_players.groupby('year')['Century'].sum()

# get the year with highest number of centuries
year_with_highest_centuries = centuries_by_year.idxmax()
year_with_highest_centuries

1998

Extracting Day From a Date
Description
The given dataset 'order' has a variable 'Order_Date' with the dates of purchase.
Create a new variable 'day' which will contain the day from the date at variable Order_Date

In [16]:
order = pd.read_csv('https://query.data.world/s/3hIAtsCE7vYkPEL-O5DyWJAeS5Af-7')
order['Order_Date'] = pd.to_datetime(order['Order_Date'])

order.head()



C:\Users\raghu\AppData\Local\Temp\ipykernel_18792\2675418305.py:2: UserWarning: Parsing dates in DD/MM/YYYY format when dayfirst=False (the default) was specified. This may lead to inconsistently parsed dates! Specify a format to ensure consistent parsing.
  order['Order_Date'] = pd.to_datetime(order['Order_Date'])


Order_ID Order_Date Order_Priority Ord_id
0         3 2010-10-13            LOW  Ord_1
1       293 2012-01-10           HIGH  Ord_2
2       483 2011-10-07           HIGH  Ord_3
3       515 2010-08-28  NOT SPECIFIED  Ord_4
4       613 2011-06-17           HIGH  Ord_5

In [17]:
order['day'] = order['Order_Date'].dt.day

In [18]:
order.head()


Order_ID Order_Date Order_Priority Ord_id  day
0         3 2010-10-13            LOW  Ord_1   13
1       293 2012-01-10           HIGH  Ord_2   10
2       483 2011-10-07           HIGH  Ord_3    7
3       515 2010-08-28  NOT SPECIFIED  Ord_4   28
4       613 2011-06-17           HIGH  Ord_5   17